crates/uv-resolver/src/resolution/display.rs

```diff
@@ -348,6 +349,28 @@ fn to_requirements_txt_graph(graph: &ResolutionPetGraph) -> IntermediatePetGraph
 /// The graph is directed, so if any edge contains a marker, we need to propagate it to all
 /// downstream nodes.
 fn propagate_markers(mut graph: IntermediatePetGraph) -> IntermediatePetGraph {
+    // Remove any cycles. By absorption, it should be fine to ignore cycles.
+    //
+    // Imagine a graph: `A -> B -> C -> A`. Assume that `A` has weight `1`, `B` has weight `2`,
+    // and `C` has weight `3`. The weights are the marker trees.
+    //
+    // When propagating, we'd return to `A` when we hit the cycle, to create `1 or (1 and 2 and 3)`,
+    // which resolves to `1`.
+    //
+    // TODO(charlie): The above reasoning could be incorrect. Consider using a graph algorithm that
+    // can handle weight propagation with cycles.
+    let edges = {
+        let fas = greedy_feedback_arc_set(&graph)
+            .map(|edge| edge.id())
+            .collect::<Vec<_>>();
+        let mut edges = Vec::with_capacity(fas.len());
+        for edge_id in fas {
+            edges.push(graph.edge_endpoints(edge_id).unwrap());
+            graph.remove_edge(edge_id);
```

Comment:
Hm is this maybe not a a StableGraph so not safe to remove edge ids like this probably, unless you sort the edge ids?

This is a drive by comment, but petgraph caught my attention (even though it doesn't do that much anymore, sorry about that :slightly_frowning_face: )

About your general questions in this PR - the back of my mind says reverse postorder to do something similar to Topo but accomodating cycles to some extent.

An algorithm like scc https://docs.rs/petgraph/latest/petgraph/algo/fn.kosaraju_scc.html  finds both the sccs (they are cyclic) as well as a topological order between each component too.

Reply:
Oh that’s a very smart comment — the edge IDs could change, right!

I would love help with this problem BTW…

Reply:
Well, I am acutely aware of petgraph's shortcomings but I'm happy that it's still useful. You managed to snipe me, so I've made a try :slightly_smiling_face: